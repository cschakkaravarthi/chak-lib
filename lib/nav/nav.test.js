"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var testHelper_1 = require("../services/testHelper");
var nav_1 = require("./nav");
var navOverlay_1 = require("./navOverlay");
var navBar_1 = require("./navBar");
var images_1 = require("../assets/images");
var breadcrumb_nav_1 = require("./breadcrumb-nav");
describe('Nav', function () {
    it('should render Breadcrumb nav', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(breadcrumb_nav_1.Breadcrumb, { onSelect: function (selectedKey) {
                return console.log("selected " + selectedKey);
            }, items: [
                {
                    key: 1,
                    isActive: true,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Home")); }
                },
                {
                    key: 2,
                    isActive: false,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Corporate Finance")); }
                },
                {
                    key: 3,
                    isActive: false,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Finance Operations")); }
                },
                {
                    key: 4,
                    isActive: false,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Local Finance Sites")); }
                }
            ] }));
        expect(tree).toMatchSnapshot();
    });
    it('should render correctly Primary Nav', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(nav_1.Nav, { items: [
                {
                    key: 1,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "News")); }
                },
                {
                    key: 2,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Employee Services")); }
                },
                {
                    key: 3,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "People & Offices")); }
                }
            ] }));
        expect(tree).toMatchSnapshot();
    });
    it('should render correctly Secondary Nav', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(nav_1.Nav, { variant: "secondary", items: [
                {
                    key: 1,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "JIRA")); }
                },
                {
                    key: 2,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Workday")); }
                },
                {
                    key: 3,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Replicon")); }
                }
            ] }));
        expect(tree).toMatchSnapshot();
    });
    it('should render correctly Secondary Nav Section', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(nav_1.Nav, { variant: 'secondary', sectionTitle: 'Quick Links', sectionLink: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Edit")); }, items: [
                {
                    key: 1,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "JIRA")); }
                },
                {
                    key: 2,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Workday")); }
                },
                {
                    key: 3,
                    Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Replicon")); }
                }
            ] }));
        expect(tree).toMatchSnapshot();
    });
    it('should render correctly nav overlay', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(navOverlay_1.NavOverlay, { show: true, editMode: false, setShow: function () { alert('clicked'); }, handleAdd: function () { alert('clicked'); }, handleEdit: function () { alert('clicked'); }, navGroups: [
                {
                    sectionTitle: 'Company',
                    sectionIcon: images_1.images.sectionIcon1,
                    variant: 'overlay',
                    items: [
                        {
                            key: 1,
                            Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Mission, Vision, and Cultural Values")); }
                        },
                        {
                            key: 2,
                            Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Culture Champion Portal")); }
                        },
                        {
                            key: 3,
                            Link: function (p) { return (react_1["default"].createElement("a", { href: "#", className: p.className }, "Leadership")); }
                        }
                    ]
                }
            ] }));
        expect(tree).toMatchSnapshot();
    });
    it('should render correctly NavBarSection', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(navBar_1.NavBar, { variant: 'top' },
            react_1["default"].createElement(navBar_1.NavBarSection, { variant: 'standard' },
                react_1["default"].createElement("p", null, "Some content in a standard section")),
            react_1["default"].createElement(navBar_1.NavBarSection, { variant: 'light' },
                react_1["default"].createElement("p", null, "Some content in a light section")),
            react_1["default"].createElement(navBar_1.NavBarSection, { variant: 'dark' },
                react_1["default"].createElement("p", null, "Some content in a dark section"))));
        expect(tree).toMatchSnapshot();
    });
    it('should render correctly navbar side', function () {
        var tree = testHelper_1.shallowRender(react_1["default"].createElement(navBar_1.NavBar, { variant: 'side' },
            react_1["default"].createElement(navBar_1.NavBarSection, { variant: 'standard' },
                react_1["default"].createElement("p", null, "Some content in a standard section")),
            react_1["default"].createElement(navBar_1.NavBarSection, { variant: 'light' },
                react_1["default"].createElement("p", null, "Some content in a light section")),
            react_1["default"].createElement(navBar_1.NavBarSection, { variant: 'dark' },
                react_1["default"].createElement("p", null, "Some content in a dark section"))));
        expect(tree).toMatchSnapshot();
    });
});
//# sourceMappingURL=nav.test.js.map