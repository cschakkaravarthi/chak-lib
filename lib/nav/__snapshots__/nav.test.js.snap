// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render Breadcrumb nav 1`] = `
<ForwardRef
  bsPrefix="nav breadcrumb-nav d-flex align-items-center bg-white undefined"
  fill={false}
  justify={false}
  onSelect={[Function]}
>
  <ForwardRef
    fluid={false}
  >
    <ForwardRef
      noGutters={false}
    >
      <ForwardRef
        bsPrefix="nav-item border-right font-weight-bold"
      >
        <Link
          className="py-1 px-4 active"
        />
      </ForwardRef>
      <ForwardRef
        bsPrefix="nav-item border-right font-weight-bold"
      >
        <Link
          className="py-1 px-4 "
        />
      </ForwardRef>
      <ForwardRef
        bsPrefix="nav-item border-right font-weight-bold"
      >
        <Link
          className="py-1 px-4 "
        />
      </ForwardRef>
      <ForwardRef
        bsPrefix="nav-item border-right font-weight-bold"
      >
        <Link
          className="py-1 px-4 "
        />
      </ForwardRef>
    </ForwardRef>
  </ForwardRef>
</ForwardRef>
`;

exports[`Nav should render correctly NavBarSection 1`] = `
<React.Fragment>
  <div
    className="nav-placeholder-top"
  />
  <div
    className="navbar-top"
  >
    <Unknown
      variant="standard"
    >
      <p>
        Some content in a standard section
      </p>
    </Unknown>
    <Unknown
      variant="light"
    >
      <p>
        Some content in a light section
      </p>
    </Unknown>
    <Unknown
      variant="dark"
    >
      <p>
        Some content in a dark section
      </p>
    </Unknown>
  </div>
</React.Fragment>
`;

exports[`Nav should render correctly Primary Nav 1`] = `
<ForwardRef
  className="nav-primary undefined"
  fill={false}
  justify={false}
>
  <ul
    className="list-unstyled mb-0"
  >
    <li>
      <Link />
    </li>
    <li>
      <Link />
    </li>
    <li>
      <Link />
    </li>
  </ul>
</ForwardRef>
`;

exports[`Nav should render correctly Secondary Nav 1`] = `
<ForwardRef
  className="nav-secondary undefined"
  fill={false}
  justify={false}
>
  <ul
    className="list-unstyled mb-0"
  >
    <li>
      <Link />
    </li>
    <li>
      <Link />
    </li>
    <li>
      <Link />
    </li>
  </ul>
</ForwardRef>
`;

exports[`Nav should render correctly Secondary Nav Section 1`] = `
<ForwardRef
  className="nav-secondary undefined"
  fill={false}
  justify={false}
>
  <div
    className="nav-section-title pt-0 font-weight-bold"
  >
    <span
      className="w-75"
    >
      Quick Links
    </span>
  </div>
  <div
    className="nav-section-link"
  >
    <sectionLink />
  </div>
  <ul
    className="list-unstyled mb-0"
  >
    <li>
      <Link />
    </li>
    <li>
      <Link />
    </li>
    <li>
      <Link />
    </li>
  </ul>
</ForwardRef>
`;

exports[`Nav should render correctly nav overlay 1`] = `
<ForwardRef(render)
  aria-labelledby="nav-overlay-title"
  backdropClassName="all-links-backdrop"
  className="nav-overlay"
  onHide={[Function]}
  show={true}
  size="xl"
>
  <ForwardRef
    closeButton={false}
    closeLabel="Close"
  >
    <ForwardRef
      className="w-100"
      noGutters={false}
    >
      <ForwardRef
        className="p-2 p-sm-2 p-md-2"
        lg={3}
        md={3}
        sm={12}
        xl={3}
      >
        <ForwardRef
          active={false}
          className="modal-close-button btn-upper text-left"
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="link"
        >
          <span
            className="css-icon-x"
          />
          Close
        </ForwardRef>
      </ForwardRef>
      <ForwardRef
        className="p-sm-2 p-md-2"
        lg={6}
        md={6}
        sm={12}
        xl={6}
      />
      <ForwardRef
        lg={3}
        md={3}
        sm={12}
        xl={3}
      >
        <Unknown
          className="btn-primary font-size-sm letter-spacing-01rem ml-2 mt-2 ml-sm-3"
          handleClick={[Function]}
          title="EDIT MY LINKS"
        />
      </ForwardRef>
    </ForwardRef>
  </ForwardRef>
  <ForwardRef>
    <ForwardRef
      noGutters={false}
    >
      <ForwardRef
        md={4}
        xl={3}
      >
        <Unknown
          items={
            Array [
              Object {
                "Link": [Function],
                "key": 1,
              },
              Object {
                "Link": [Function],
                "key": 2,
              },
              Object {
                "Link": [Function],
                "key": 3,
              },
            ]
          }
          sectionIcon="test-file-stub"
          sectionTitle="Company"
          variant="secondary"
        />
      </ForwardRef>
    </ForwardRef>
  </ForwardRef>
</ForwardRef(render)>
`;

exports[`Nav should render correctly navbar side 1`] = `
<React.Fragment>
  <div
    className="navbar-side"
  >
    <Unknown
      variant="standard"
    >
      <p>
        Some content in a standard section
      </p>
    </Unknown>
    <Unknown
      variant="light"
    >
      <p>
        Some content in a light section
      </p>
    </Unknown>
    <Unknown
      variant="dark"
    >
      <p>
        Some content in a dark section
      </p>
    </Unknown>
  </div>
</React.Fragment>
`;
